services/api/src/bitbucket-sync/repo-permissions.ts

```diff
@@ -61,108 +61,160 @@ const getBitbucketRepo = async (gitUrl: string, projectName: string) => {
-
-    let userPermissions = [];
-    try {
-      const permissions = await bitbucketApi.getRepoUsers(bbProject, bbRepo);
-
-      // Useres w/o an email address were deleted/deactivated, but somehow
-      // still returned in the API
-      // @ts-ignore
-      userPermissions = R.filter(R.propSatisfies(R.has('emailAddress'), 'user'), permissions);
-    } catch (e) {
-      logger.warn(`Could not load users for repo ${R.prop('slug', repo)}: ${e.message}`);
-      continue;
-    }
-
-    // Sync user/permissions from bitbucket to lagoon
-    for (const userPermission of userPermissions) {
-      const bbUser = userPermission.user as BitbucketUser;
-      const bbPerm = userPermission.permission;
-
-      try {
-        const email = bbUser.emailAddress.toLowerCase();
+          const bbProject = R.path(['project', 'key'], repo);
+          const bbRepo = R.prop('slug', repo);
+          logger.debug(
+            `Processing project "${projectName}", bitbucket "${bbRepo}"`
+          );
+
+          let userPermissions = [];
+
+          const lagoonUsersInGroup = await getLagoonUsersForGroup(
+            lagoonProjectGroup
+          );
+
+          try {
+            const permissions = await bitbucketApi.getRepoUsers(
+              bbProject,
+              bbRepo
+            );
+
+            // Useres w/o an email address were deleted/deactivated, but somehow
+            // still returned in the API
+            // @ts-ignore
+            userPermissions = R.filter(
+              R.propSatisfies(R.has('emailAddress'), 'user'),
+              permissions
+            );
+          } catch (e) {
+            logger.warn(
+              `Could not load users for repo ${R.prop('slug', repo)}: ${
+                e.message
+              }`
+            );
+            return;
+          }
 
-        if (!R.contains(email, existingUsers)) {
-          const userAddedOrExists = await addUser(email);
-          if (!userAddedOrExists) {
-            // Errors for this case are logged in addUser
-            continue;
+          // Sync user/permissions from bitbucket to lagoon
+          for (const userPermission of userPermissions) {
+            const bbUser = userPermission.user as BitbucketUser;
+            const bbPerm = userPermission.permission;
+
+            try {
+              const email = bbUser.emailAddress.toLowerCase();
+
+              const userExistsInGroupAlready = !R.contains(
+                email,
+                lagoonUsersInGroup
+              );
+
+              if (
+                !R.contains(email, existingUsers) &&
+                !userExistsInGroupAlready
+              ) {
+                const userAddedOrExists = await addUser(email);
+                if (!userAddedOrExists) {
+                  // Errors for this case are logged in addUser
+                  continue;
+                }
+
+                existingUsers.push(email);
+              }
+              if (!userExistsInGroupAlready) {
```

Comment:
Hmm, I just realized this might need to go/change - it might be too aggressively ignoring users. I need to check if their permissions have changed.

Reply:
Removed this in case, still way faster.